simpleclient_spring_web/src/main/java/io/prometheus/client/spring/web/MethodTimer.java

```diff
@@ -0,0 +1,96 @@
+package io.prometheus.client.spring.web;
+
+import io.prometheus.client.Summary;
+import org.apache.commons.lang3.StringUtils;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+
+import java.util.HashMap;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * This class automatically times (via aspectj) the execution of methods by their signature, if it's been enabled via {@link EnablePrometheusTiming}
+ * and in methods annotated (or within classes annotated) with {@link PrometheusTimeMethods}
+ *
+ * @author Andrew Stuart
+ */
+@Aspect
+@ControllerAdvice
+@Component
+public class MethodTimer {
+    public static final String METRIC_NAME = "prometheus_method_timing_seconds";
+
+    public static final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+    public static final HashMap<String, Summary> collectors = new HashMap<String, Summary>(10);
+
+    public static final Summary defaultSummary = Summary.build()
+            .name(METRIC_NAME)
+            .help("Automatic annotation-driven method timing")
+            .labelNames("signature")
+            .register();
+
+    private Summary getSummary(PrometheusTimeMethods annot) {
+        Summary summary;
+        String name = annot.value();
```

Comment:
I guess I'll have to read up on Spring AOP a little more.  My understanding was that it's typically a singleton handling the methods, but it appears there can be advice instances. Any specific suggestions on implementation there?